src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+    const arrayLength = this.length;
+
+    for (let i = 0; i < arrayLength; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        string += this[i];
+      }
```

Comment:
A good place to use [the nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)